pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -963,6 +966,272 @@ func (a *ApplicationBackupController) prepareResources(
 	return nil
 }
 
+func (a *ApplicationBackupController) updateRancherProjectDetails(
+	backup *stork_api.ApplicationBackup,
+	objects []runtime.Unstructured,
+) error {
+	platformCredential, err := storkops.Instance().GetPlatformCredential(backup.Spec.PlatformCredential, backup.Namespace)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	if platformCredential.Spec.Type == stork_api.PlatformCredentialRancher {
+		if err = updateRancherProjects(platformCredential, backup, objects); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func updateRancherProjects(
+	platformCredential *stork_api.PlatformCredential,
+	backup *stork_api.ApplicationBackup,
+	objects []runtime.Unstructured,
+) error {
+	projects := map[string]string{}
+	for _, o := range objects {
+		metadata, err := meta.Accessor(o)
+		if err != nil {
+			return err
+		}
+		resource := o.GetObjectKind().GroupVersionKind()
+		switch resource.Kind {
+		case "Deployment", "StatefulSet", "DeploymentConfig", "IBPPeer", "IBPCA", "IBPConsole", "IBPOrderer", "ReplicaSet":
+			err := getProjectsFromPodNamespaceSelector(o, projects)
+			if err != nil {
+				return fmt.Errorf("error preparing %v resource %v: %v", o.GetObjectKind().GroupVersionKind().Kind, metadata.GetName(), err)
+			}
+		case "NetworkPolicy":
+			err := getProjectsFromNetworkPolicy(o, projects)
+			if err != nil {
+				return fmt.Errorf("error preparing %v resource %v: %v", o.GetObjectKind().GroupVersionKind().Kind, metadata.GetName(), err)
+			}
+		case "PersistentVolume":
+			err := getProjectsFromPV(o, projects)
+			if err != nil {
+				return fmt.Errorf("error preparing %v resource %v: %v", o.GetObjectKind().GroupVersionKind().Kind, metadata.GetName(), err)
+			}
+		case "PersistentVolumeClaim":
+			err := getProjectsFromPVC(o, projects)
+			if err != nil {
+				return fmt.Errorf("error preparing %v resource %v: %v", o.GetObjectKind().GroupVersionKind().Kind, metadata.GetName(), err)
+			}
+		}
+	}
+
+	for _, namespace := range backup.Spec.Namespaces {
+		ns, err := core.Instance().GetNamespace(namespace)
+		if err != nil {
+			return err
+		}
+		for key, val := range ns.Annotations {
+			if strings.Contains(key, utils.CattleProjectPrefix) {
+				projects[val] = ""
+			}
+		}
+		for key, val := range ns.Labels {
```

Comment:
As i understand when a NS is created and added to project, both annotation and Labels should have same project name. Do we foresee any case where these values are different?

Reply:
Yes both the annotation and label should be pointing to the same project.
If you create the namespace through management UI, the label and annotation for projectID is as given below:
```
labels:
    field.cattle.io/projectId: p-62xs2
```
```
annotation: ( In annotation, it contains the clusterID:projectID)
field.cattle.io/projectId: c-xg2ls:p-62xs2
```   
Some times, if it is created by applying the yaml directly, it is possible, it contains only one of them. So we are taking the value, if it is present in either in label or annotation.
@px-kesavan, But one thing, I noticed during the test is that if we have only label and not annotation, the management UI does not associate the namespace to a project, We need to have annotation to associate the ns to a project. Label is for just informational. So I feel, we should pick up the project only when it is present in the annotation.
For now, I will remove the label loop.

Reply:
@prashanthpx @siva-portworx In the namespace, both annotation and label will have the same projectID unless someone removes or modifies the annotation manually. 
If namespace is again move to proper project from UI , the rancher controller will automatically updates the annotation and label with correct values. 
if the namespace has valid projectID, the rancher controller will automatically update the label. 

Yes, we can remove the label check for namespace and get it from annotation .

Reply:
removed the label loop